torch/csrc/jit/python/pybind_utils.cpp

```diff
@@ -69,18 +70,23 @@ IValue toIValue(py::handle obj, const TypePtr& type, c10::optional<int32_t> N) {
       size_t tuple_size = tuple.size();
       auto tuple_type = type->cast<TupleType>();
       const auto& elem_types = tuple_type->elements();
-      if (elem_types.size() != tuple_size) {
```

Comment:
Why is this check removed?

Reply:
This is because annotation like Tuple[Tensor, Optional[Tensor]] would think the tuple size is 2 while python thinks it is size of 1 (e.g (torch.rand(4),). My idea was instead of checking size here, we just need to make sure that we are inserting None.